src/jmespath.net/JmesPath.cs

```diff
@@ -7,13 +7,44 @@
 using DevLab.JmesPath.Interop;
 using DevLab.JmesPath.Utils;
 using Newtonsoft.Json;
+using System.Collections.Generic;
 
 namespace DevLab.JmesPath
 {
+    public sealed class __ : IScopeParticipant, IContextEvaluator

```

Comment:
Find a new home and proper name for this variable.

Reply:
ScopeVisitor or ScopeEvaluator ?

Reply:
Named `ScopeParticipantVisitor`.